### DISCUSSION TITLE
Websocket .emit is not a function AdonisJS 4.1

### DISCUSSION THREAD

**Discussion by 13Aks13** [adonisjs/core#1578] [Help]
What's wrong?  Why .emit is not a function? 

'use strict'

const Ws = use('Ws')

class SocketHelper {

  static sendEvent(channel, message, data, typeEvent = 0) {
    const socket = Ws.getChannel(channel).topic(channel)
    if (socket) {
      /*
        EMIT = 0,
        BROADCAST = 1,
        BROADCASTALL = 2;
       */
       if (typeEvent == 1) {
         try {
           socket.broadcast(message, data)
         } catch(error) {
           console.error("Socket send error: ", error)
         }
       } else if (typeEvent == 2) {
         try {
           socket.broadcastToAll(message, data)
         } catch(error) {
          console.error("Socket send error: ", error)
         }
       } else {
         try {
          socket.emit(message, data)
         } catch(error) {
          console.error("Socket send error: ", error)
         }
       }
    }
  }

}

module.exports = SocketHelper

**Comment by yarapolana** [comment]
Hey @13Aks13 
probably `const socket = Ws.getChannel(channel).topic(channel)` is coming up undefined
you could tweak your helper like this
hope this helps

  ```
const channel = Ws.getChannel('room:*')
  if (!channel) return

  const topic = channel.topic(`room:${id}`)
  if (!topic) {
    console.error('Has no topic')
    return
  }

  // emit, broadcast, broadcastToAll
  topic.broadcastToAll('message', {
    type,
    data,
  })
```
